docs/TEST_GUIDE.md

```diff
@@ -0,0 +1,75 @@
+[middleware](/LiskHQ/lisk-hub/blob/development/src/store/middlewares), or
+[subscriber](/LiskHQ/lisk-hub/blob/development/src/store/subscribers)) in isolation.
+
+### How are they organized?
+Each unit test live in the same folder as the unit that it tests. E.g. tests for [/src/components/login/login.js](/LiskHQ/lisk-hub/blob/development/src/components/login/login.js) are in [/src/components/login/login.test.js](/LiskHQ/lisk-hub/blob/development/src/components/login/login.test.js).
+
+Configuration is in [karma.conf.js](/LiskHQ/lisk-hub/blob/development/karma.conf.js) and [config/webpack.config.test.js](/LiskHQ/lisk-hub/blob/development/config/webpack.config.test.js).
+
+### How to run them?
+See [relevant sections of README](/LiskHQ/lisk-hub#run-unit-tests)
+
+### What tools are used?
+- **Assertions** use `expect` syntax of [chai](http://www.chaijs.com/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/80afda8289b75cb70bf345d14d36117fde2bdd90/src/utils/passphrase.test.js#L136
+
+- **Spies, stubs, ands mocks** use [sinon](http://sinonjs.org/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/19a2a613fa08a34fe3088b0e40c11c7fa37a645d/src/store/middlewares/savedSettings.test.js#L20
+
+- To test **React compoments** we use [enzyme](http://airbnb.io/enzyme/), e.g.:
+https://github.com/LiskHQ/lisk-hub/blob/79165170a326a7f98efee098732e55be37d31223/src/components/toolbox/sliderCheckbox/index.test.js#L21
+
+
+## Integration tests
+
+### What do they test?
+Integration of all components, utils, reducers and middlewares on one page, e.g. [Login](/LiskHQ/lisk-hub/blob/development/test/integration/login.test.js), [Wallet](/LiskHQ/lisk-hub/blob/development/test/integration/wallet.test.js), [Delegates](/LiskHQ/lisk-hub/blob/development/test/integration/voting.test.js).
+
+### How are they organized?
+Integration tests for each page have their own `*.test.js` file in [/test/integration/](/LiskHQ/lisk-hub/blob/development/test/integration).
+
+Configuration is common with unit tests.
+
+### How to run them?
+They are run together with unit tests.
+
+### What tools are used?
+- All that are used for unit tests.
+- [mocha-steps](https://www.npmjs.com/package/mocha-steps) to write the tests in a more [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development) way.
+- Our own **generic step definitions** [/test/utils/genericStepDefinition.js](LiskHQ/lisk-hub/blob/bfc94e4f46b4e2393bcc1a0ecd6f1bc85590b6a6/test/utils/genericStepDefinition.js)
+- Our own **mount helper** that wraps `enzyme.mount` to avoid some code repetition: [/test/utils/mountHelpers.js](https://github.com/LiskHQ/lisk-hub/blob/bfc94e4f46b4e2393bcc1a0ecd6f1bc85590b6a6/test/utils/mountHelpers.js)
+
+
+## End-to-end (E2E) tests
+
+### What do they test?
+Full user scenarios in the application as a whole, including the communication with Lisk Core. 
+
+### How to run them?
+See [relevant sections of README](/LiskHQ/lisk-hub#run-end-to-end-tests)
+
+### How are they organized?
```

Comment:
Just like the other 2, I think we need to define boundaries of e2e tests scenarios so we don't end up having a scenario like:
  

-   Given I'm on login page
-   When I fill in passphrase of "genesis" to "passphrase" field
-   And I click "login button"
-   Then I should be logged in as "genesis" account